data/transactions/logic/eval.go

```diff
@@ -1866,9 +1875,10 @@ func opBytesEq(cx *EvalContext) error {
 		return errors.New("math attempted on large byte-array")
 	}
 
-	rhs := new(big.Int).SetBytes(cx.stack[last].Bytes)
-	lhs := new(big.Int).SetBytes(cx.stack[prev].Bytes)
-	cx.stack[prev] = boolToSV(lhs.Cmp(rhs) == 0)
+	rhs := nonzero(cx.stack[last].Bytes)
+	lhs := nonzero(cx.stack[prev].Bytes)
+
+	cx.stack[prev] = boolToSV(bytes.Equal(lhs, rhs))
```

Comment:
does bytes.Equal (nil, nil) work the same way as big.Int.Cmp?

Reply:
Yes: https://go.dev/play/p/2ekrBRqByzq

But I'll add some more unit tests that use all zero byte inputs to demonstrate.